File: problems/198. House Robber/solution.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if not nums:
            return 0
        if len(nums) == 1:
            return nums[0]
        sums = [0 for _ in nums]
        sums[0] = nums[0]
        sums[1] = max(nums[0], nums[1])

        for i in range(2, len(nums)):
            sums[i] = max(nums[i] + sums[i-2], sums[i-1])
        return max(sums)

File: problems/198. House Robber/solution.cpp

class Solution
{
public:
    int rob(vector<int> &nums)
    {
        if (nums.size() == 0)
            return 0;
        if (nums.size() == 1)
            return nums[0];
        std::vector<int> result(nums.size());
        result[0] = nums[0];
        result[1] = std::max(nums[0], nums[1]);

        for (int i = 2; i < nums.size(); i++)
        {
            result[i] = std::max(nums[i] + result[i - 2], result[i - 1]);
        }
        std::vector<int>::iterator it;
        it = max_element(result.begin(), result.end());
        return *it;
    }
};